### PR TITLE
[KW-Search] Create `data_sources_nodes` and `folders` tables

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
Fixes https://github.com/dust-tt/tasks/issues/1609

`node_id` duped from linked items as discussed in [thread](https://dust4ai.slack.com/archives/C050A0S2Z7F/p1729868140336129)

## Risk
none, no operations on the tables done yet

## Deploy Plan
migration
deploy core

### CODE REVIEW

File: core/src/stores/store.rs [hunk 1]
```diff
@@ -486,9 +497,40 @@ pub const POSTGRES_TABLES: [&'static str; 14] = [
        remote_database_secret_id    TEXT,
        FOREIGN KEY(data_source)     REFERENCES data_sources(id)
     );",
+    "-- data sources folders
+    CREATE TABLE IF NOT EXISTS data_sources_folders (
+       id                           BIGSERIAL PRIMARY KEY,
+       data_source                  BIGINT NOT NULL,
+       created                      BIGINT NOT NULL,
```
Comment (spolu) [inline core/src/stores/store.rs:504]: nit move created up

File: core/src/stores/store.rs [hunk 2]
```diff
@@ -486,9 +497,40 @@ pub const POSTGRES_TABLES: [&'static str; 14] = [
...
+    );",
+    "-- data sources nodes
+    CREATE TABLE IF NOT EXISTS data_sources_nodes (
+       id                           BIGSERIAL PRIMARY KEY,
+       data_source                  BIGINT NOT NULL,
+       created                      BIGINT NOT NULL,
+       timestamp                    BIGINT NOT NULL,
+       node_id                      TEXT NOT NULL,
+       title                        TEXT NOT NULL,
+       mimeType                     TEXT NOT NULL,
```
Comment (spolu) [inline core/src/stores/store.rs:516]: snake_case

File: core/src/stores/postgres.rs [hunk 3]
```diff
@@ -2844,6 +2846,185 @@ impl Store for PostgresStore {
...
+                    title,
+                    mime_type,
+                    parents,
+                }))
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    async fn delete_data_source_node(&self, data_source_id: &str, node_id: &str) -> Result<()> {
```
Comment (spolu) [inline core/src/stores/postgres.rs:3004]: Are we really going to use that? shouldn't it be embedded in the deletion of documents/tables/folders

**Review by spolu — APPROVED** [review]
Left a few comments but LGTM otherwise!

File: core/src/stores/postgres.rs (hunk 3 above)
Comment (philipperolet) [reply core/src/stores/postgres.rs:3004]: I thought postgres.rs was for atomic operations, and logic like that was to be a notch above, in the callers of postgres methods

Moving it in deletion is ~resourcy. I think it makes sense, it's pit-of-successy but then we should move upsertion too, no? inside upsertion of docs/tables/folders

wdyt?

Comment (spolu) [reply core/src/stores/postgres.rs:3004]: Yes we might and we'll figure it out as we go 👍

Fine to keep these functions and we'll turn them in functions that we can be called with a transaction from other functions here 👍

**Pull request merged**